base/src/main/java/org/mozilla/jss/provider/java/security/KeyFactorySpi1_2.java

```diff
@@ -77,45 +83,46 @@ protected PublicKey engineGeneratePublic(KeySpec keySpec)
-        //   // We need to import both the public value and the curve.
-        //   // The only way to get all that information in DER is to send
-        //   // a full SubjectPublicKeyInfo. So we encode all the information
-        //   // into an SPKI.
-        //
-        //  ECPublicKeySpec spec = (ECPublicKeySpec) keySpec;
-	//    AlgorithmParameters algParams = getInstance("ECParameters");
-        //
-        //    algParameters.init(spec.getECParameters());
-        //    OBJECT_IDENTIFIER oid = null;
-        //    try {
-        //        oid = SignatureAlgorithm.ECSignature.toOID();
-        //    } catch(NoSuchAlgorithmException ex ) {
-        //        Assert.notReached("no such algorithm as DSA?");
-        //    }
-        //    AlgorithmIdentifier algID =
-        //                  new AlgorithmIdentifier(oid, ecParams.getParams() );
-        //    INTEGER publicValueX = new INTEGER(spec.getW().getAffineX());
-        //    INTEGER publicValueY = new INTEGER(spec.getW().getAffineY());
-        //    byte[] encodedPublicValue;
-        //    encodedPublicValue[0] = EC_UNCOMPRESSED_POINT;
-        //    encodedPublicValue += spec.getW().getAffineX().toByteArray();
-        //    encodedPublicValue += spec.getW().getAffineY().toByteArray();
-        //
-        //    byte[] encodedPublicValue = ASN1Util.encode(publicValue);
-        //    SubjectPublicKeyInfo spki = new SubjectPublicKeyInfo(
-        //        algID, new BIT_STRING(encodedPublicValue, 0) );
-        //
-        //   return PK11PubKey.fromSPKI( ASN1Util.encode(spki) );
-        //
-        // use the following for EC keys in 1.4.2
+        } else if(keySpec instanceof ECPublicKeySpec ecPKSpec) {
+
+            ECCurve curve = ECCurve.fromOrder(ecPKSpec.getParams().getOrder());
+            if (curve == null) {
+                throw new InvalidKeySpecException("Unsupported Elliptic Curve");
+            }
+            int pointSize = switch(curve) {
+                case P256 -> 32;
+                case P384 -> 48;
+                case P521 -> 66;
+                default -> 32;
+            };
+
+            OBJECT_IDENTIFIER oid = null;
+            try {
+                oid = SignatureAlgorithm.ECSignature.toOID();
+            } catch(NoSuchAlgorithmException ex ) {
+                throw new InvalidKeySpecException("Unsupported KeySpec type: " +
+                                                  keySpec.getClass().getName());
+            }
+			
+            AlgorithmIdentifier algID =
+                new AlgorithmIdentifier(oid, curve.getOIDs()[0]);
+
+            byte[] publicValueX = convertIntToBigEndian(ecPKSpec.getW().getAffineX(), pointSize);
+            byte[] publicValueY = convertIntToBigEndian(ecPKSpec.getW().getAffineY(), pointSize);
+
+            byte[] encodedPublicValue = new byte[1 + publicValueX.length + publicValueY.length];
+            encodedPublicValue[0] = 0x04; //EC_UNCOMPRESSED_POINT
```

Comment:
I agree that this block could be improved with ArrayCopy since we seem to be concatenating two byte arrays together with the tag byte at the front.